test/jest/roktManager.spec.ts

```diff
@@ -0,0 +1,119 @@
+import RoktManager, { IRoktLauncher, ISelectPlacementsOptions } from "../../src/roktManager";
+
+describe('RoktManager', () => {
+    let roktManager: RoktManager;
+
+    beforeEach(() => {
+        roktManager = new RoktManager();
+    });
+
+    describe('constructor', () => {
+        it('should be initialized', () => {
+            expect(roktManager).toBeDefined();
+        });
+        it('should have a null launcher', () => {
+            expect(roktManager['launcher']).toBeNull();
+        });
+    });
+
+    describe('#attachLauncher', () => {
+        it('should attach a launcher', () => {
+            const launcher = {} as IRoktLauncher;
+            roktManager.attachLauncher(launcher);
+            expect(roktManager['launcher']).not.toBeNull();
+        });
+
+        it('should queue the launcher method if no launcher is attached', () => {
+            roktManager.attachLauncher(null);
+            expect(roktManager['launcher']).toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(1);
+            expect(roktManager['messageQueue'][0].methodName).toBe('attachLauncher');
+            expect(roktManager['messageQueue'][0].payload).toBe(null);
+        });
+
+        it('should process the message queue if a launcher is attached', () => {
+            const launcher = jest.fn() as unknown as IRoktLauncher;
+
+            roktManager.attachLauncher(null);
+            expect(roktManager['launcher']).toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(1);
+            expect(roktManager['messageQueue'][0].methodName).toBe('attachLauncher');
+            expect(roktManager['messageQueue'][0].payload).toBe(null);
+
+            roktManager.attachLauncher(launcher);
+            expect(roktManager['launcher']).not.toBeNull();
+            expect(roktManager['messageQueue'].length).toBe(0);
+        });
+    });
+
+    describe('#selectPlacements', () => {
+        it('should call selectPlacements', () => {
```

Comment:
in general for all these tests, i think it gets confusing when you are saying selectPlacements gets called.  we should be more explicit that one `selectPlacements` gets mapped to anotheri `selectPlacements` call think being explicit is good here.  `should call launcher.selectPlacements when roktManager.selectPlacements is called`

```suggestion
        it('should map a IRoktLauncher selectPlacement to a launcher selectPlacements call', () => {
```

Reply:
I think "map an IRoktLauncher selectPlacement" is too much of an implementation detail for this specific tests, but I agree that it should more explicit to referenc the launcher. I'll revise.